src/opustrainer/trainer.py

```diff
@@ -199,11 +199,18 @@ def __next__(self):
         assert self._fh is not None
         try:
             # Try to read the next line from our shuffled file
-            line = self._fh.readline()
-            if line == '':
-                raise StopIteration
-            self.line += 1
-            return line
+            while True:
+                line = self._fh.readline()
+                if line == '':
+                    raise StopIteration
+                self.line += 1
+                # assert that the line is well formed, meaning non of the fields is the empty string
+                # If not, try to get a new line from the corpus
+                testline: List[str] = line.rstrip('/r/n').strip().split('/t')
```

Comment:
Should be `\` instead of `/`. Also the `strip()` will mess things up here because empty columns will be stripped entirely, yielding just a row with fewer fields. Only when a column that is surrounded by non-empty columns is empty, this will be found.

Reply:
:facepalm: